From 829da61c0229c0734a109339f5de308bf8205bf6 Mon Sep 17 00:00:00 2001
From: Arne Fitzenreiter <arne.fitzenreiter@ipfire.org>
Date: Wed, 6 Jun 2012 08:06:15 +0200
Subject: [PATCH] bcm2835-ctl: limit maximal volume to 4db.

it makes no sense to set 23.04db as maximum volume since around 3db it start to cliping. So with 4db the alsamixer is much better to control. (86% is 0db)
---
 sound/arm/bcm2835-ctl.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/arm/bcm2835-ctl.c b/sound/arm/bcm2835-ctl.c
index 828a820..f901ddd 100755
--- a/sound/arm/bcm2835-ctl.c
+++ b/sound/arm/bcm2835-ctl.c
@@ -53,7 +53,7 @@ static int snd_bcm2835_ctl_info(struct snd_kcontrol *kcontrol,
 		uinfo->type = SNDRV_CTL_ELEM_TYPE_INTEGER;
 		uinfo->count = 1;
 		uinfo->value.integer.min = -10240;
-		uinfo->value.integer.max = 2303;
+		uinfo->value.integer.max = 400;      /* 2303 */
 	} else if (kcontrol->private_value == PCM_PLAYBACK_MUTE) {
 		uinfo->type = SNDRV_CTL_ELEM_TYPE_INTEGER;
 		uinfo->count = 1;
-- 
1.7.10

